test/test.js

```diff
@@ -458,6 +462,43 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+      pathname === '/browserWorkerTestReports') {
+
+    var writableStream = fs.createWriteStream(
+                           '../coverage/coverageinfo.json');
+    var body = '';
+
+    req.on('data', function(chunk) {
+      body += chunk;
+      writableStream.write(chunk);
```

Comment:
You should not write the file until the response has been received and merged. Otherwise the JSON file will end up having only the data for one coverage report (in the best case), and broken JSON in the worst case (when at least two coverage reports are being written simultaneously).

Reply:
So maybe we'll have to write the entire body at one go ?
That too when it has been received by either of them ?